python_modules/dagster/dagster/_core/test_utils.py

```diff
@@ -689,3 +690,13 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def raise_exception_on_warnings():
+    # turn off any outer warnings filters, e.g. ignores that are set in pyproject.toml
+    warnings.resetwarnings()
+    warnings.filterwarnings("error")
+
+    if sys.version_info >= (3, 12):
+        # pendulum sometimes raises DeprecationWarning on python3.12
+        warnings.filterwarnings("ignore", category=DeprecationWarning, module="pendulum")
```

Comment:
unrelated pendulum DeprecationWarnings were failing tests that tried to suppress dagster ExperimentalWarnings